src/components/calcite-dropdown/calcite-dropdown.tsx

```diff
@@ -409,10 +409,6 @@ export class CalciteDropdown {
     ) {
       if (target.nodeName !== "BUTTON" && target.nodeName !== "CALCITE-BUTTON") {
```

Comment:
Can we scope this fix to the calcite-button? As it stands this removes the ability for any other element (`calcite-link`, a `div`, etc.) to trigger the dropdown via keyboard.

Reply:
I think this is probably out of the scope of this PR but I agree, this code looks suspicious. We shouldn't be checking node types here.

Reply:
This one also: `const itemToFocus = target.nodeName !== "A" ? target : target.parentNode;`